src/helpers/sharingHelper.js

```diff
@@ -485,4 +485,35 @@ module.exports = {
       throw new Error("Invalid Share Type" + lastShare);
     }
   },
+  /**
+   * updates the shared file/folder permission by sharer
+   *
+   * @param shauser
+   * @param resource
+   * @param permissionsreTypeString
+   */
+  updateSharedFilePermissionByUser: async function (
+    sharer,
+    resource,
+    permissions
+  ) {
+    let shareID;
+    const shares = await this.getAllSharesSharedByUser(sharer);
+    for (const shareElement of shares) {
+      if (shareElement.uid_owner === sharer && shareElement.path === resource) {
+        shareID = shareElement.id.toString();
+        break;
+      }
+    }
+    if (!shareID) {
+      return Promise.reject(new Error("Could not get shareId for last share"));
```

Comment:
this function does not know anything about "last share", so this error message is not so helpful.
```suggestion
      return Promise.reject(new Error("Could not find '" . resource . "' in shares, shared by '" . sharer . "'"));
```